lib/queue.rb

```diff
@@ -1,31 +1,63 @@
 class Queue
 
+# Using a circular buffer with an internal array starting at 20 elements, implement a Queue
+
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @size = 20
+    @front = -1
+    @back = -1
   end
 
-  def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+  # Adds the value to the back of the queue.
+  def enqueue(value)
+    if @front == -1
+      @front = 0
+      @back = 1
+      @store[@front] = value
+    elsif @store.empty?
+      raise ArgumentError, "Queue is full"
```

Comment:
Note about the readability here.  You raise a Queue is full error if the array is empty?